File: server/src/main/java/ee/rsx/kata/bank/loans/domain/LoanEligibility.java
package ee.rsx.kata.bank.loans.domain;

import ee.rsx.kata.bank.loans.api.eligibility.LoanEligibilityStatus;
import jakarta.annotation.Nullable;

public record LoanEligibility(
  LoanEligibilityStatus status,
  @Nullable Integer eligibleAmount,
  @Nullable Integer eligiblePeriod
) {}

File: server/loans/core/src/main/kotlin/ee/rsx/kata/bank/loans/domain/limits/gateway/LoanEligibility.kt
package ee.rsx.kata.bank.loans.domain.limits.gateway

import ee.rsx.kata.bank.loans.eligibility.LoanEligibilityStatus

data class LoanEligibility(
  val status: LoanEligibilityStatus,
  val eligibleAmount: Int? = null,
  val eligiblePeriod: Int? = null
)
